doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,157 @@
+| CMIS v5 | [CMIS5p0.pdf](http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf) |
+
+
+# Problem Definition
+
+1.	Presently in SONiC, there is no synchronization between 400G Datapath Init operation and enabling ASIC (NPU/PHY) Tx which may cause link instability during administrative interface enable “config interface startup Ethernet” configuration and bootup scenarios. 
+      
+    For 400G optics module, the Host (NPU/PHY) needs to provide a valid high-speed Tx input signal at the required signaling rate and encoding type prior to causing a DPSM to exit from DPDeactivated state and to move to DP Init transient state.
+      
+    Fundamentally it means - have a deterministic approach to bring-up the interface.
+      
+    Also, this problem is mentioned ‘as outside-the-scope’ of ‘CMIS Application Initialization’ high-level design document
+      **(https://github.com/ds952811/SONiC/blob/0e4516d7bf707a36127438c7f2fa9cc2b504298e/doc/sfp-cmis/cmis-init.md#outside-the-scope)**
+
+2.  During administrative interface disable “config interface shutdown Ethernet”, only the ASIC(NPU) Tx is disabled and not the optics laser for 100G/400G. 
+      This will lead to power wastage and un-necessary fan power consumption to keep the module temperature in operating range 
+
+# Background
+
+  Per the ‘QSFPDD spec’,  ‘validation, diagnostics’ done by HW team' and 'agreement with vendors', 
+  need to follow following bring-up seq to enable port/interface with 400G optics in LC/chassis:
+
+    a) Enable port on NPU (bring-up port, serdes on the NPU ; enable signals) : syncd
+    b) Enable port on PHY (bring-up port, serdes on the PHY ; enable signals) : gbsyncd
+       - Wait for signal to stabilize on PHY   
+    c) Enable optical module (turn laser on/ enable tx) : xcvrd or platform bootstrap/infra
+
+  In boards not having PHY, #b) not needed but #a) and #c) sequence to be followed.
+  
+  ## Clause from QSFP-DD (CMIS4.0 spec)
+  
+     Excerpt from CMIS4.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![61f5b485-cf3b-4ca8-beac-9102b6feabfe](https://user-images.githubusercontent.com/69485234/147173702-f124fc9d-ef27-4816-b1a1-b4a44a5833a7.PNG)
+
+
+  ## Clause from QSFP-DD (CMIS5.0 spec)
+  
+     Excerpt from CMIS5.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![96a35dc5-618f-418c-9593-5639a90f1b28](https://user-images.githubusercontent.com/69485234/147173164-5ad0123c-479a-4774-b3ee-12a81fdd7d7e.PNG)
+     
+
+# Objective
+
+Have a determistic approach for Interface link bring-up sequence i.e. below sequence to be followed:
+  1. Initialize and enable NPU Tx and Rx path
+  2. For system with 'External' PHY: Initialize and enable PHY Tx and Rx on both line and host sides; ensure host side link is up 
+  3. Then only perform optics data path initialization/activation/Tx enable (for 400G) and Tx enable (for 100G) 
```

Comment:
recommend CMIS-compliant instead of 400G and SFF-8636 (if that's right) for 100G since CMIS-compliant optics can be run at 100G (either via channelization or running at a lower line rate) and would still need to follow the CMIS init sequence

Reply:
@mikeberesford 
Sure, I will change the 400G with CMIS and 100G with SFF here.